cmd/gateway-azure.go

```diff
@@ -310,90 +310,82 @@ func (a *azureObjects) DeleteBucket(bucket string) error {
 // ListObjects - lists all blobs on azure with in a container filtered by prefix
 // and marker, uses Azure equivalent ListBlobs.
 func (a *azureObjects) ListObjects(bucket, prefix, marker, delimiter string, maxKeys int) (result ListObjectsInfo, err error) {
-	resp, err := a.client.ListBlobs(bucket, storage.ListBlobsParameters{
-		Prefix:     prefix,
-		Marker:     marker,
-		Delimiter:  delimiter,
-		MaxResults: uint(maxKeys),
-	})
-	if err != nil {
-		return result, azureToObjectError(traceError(err), bucket, prefix)
-	}
-	result.IsTruncated = resp.NextMarker != ""
-	result.NextMarker = resp.NextMarker
-	for _, object := range resp.Blobs {
-		if strings.HasPrefix(object.Name, globalMinioSysTmp) {
-			continue
+	var objects []ObjectInfo
+	var prefixes []string
+	for len(objects) == 0 && len(prefixes) == 0 {
```

Comment:
Issue: We may give lesser than `maxKeys` objects matching `prefix` starting after `marker` as we break the `for` loop once we have non-zero `objects` or `prefixes`.

Possible solution:
```go
for len(objects) < maxKeys {
...
}
```

Reply:
This is indeed an expected behavior i.e. we return less items than requested max keys with `IsTruncated = true`.

Returning 1000 keys with `IsTruncated - True` would lead to complicated logic.

Reply:
@balamurugana this makes sense.

Reply:
This is not true with our server mode such as FS and Erasure Code, is the agreed across all gateway implementations?

Reply:
Unable to understand.  Could you tell me the problem exactly?

Reply:
@balamurugana i meant in FS and Erasure code mode we return exact 1000 if max-keys=1000 . Are you saying that Azure will return 999 instead?